crates/interpreter/src/valid.rs

```diff
@@ -412,31 +423,42 @@ struct Expr<'a, 'm> {
     side_table: SideTable,
 }
 
+#[allow(dead_code)]
+#[derive(Default)]
+struct ParserRange {
+    start: usize,
+    end: usize, // exclusive
+}
+
 #[derive(Default)]
 struct SideTable {
```

Comment:
We should rename this to `FuncMeta` since we now generalized the definition of side table.